public/docs/ts/latest/guide/style-guide.jade

```diff
@@ -0,0 +1,651 @@
+    SpeakerDetailComponent  //speaker-detail.component.ts
+    ```
+
+<a href="#toc">Back to top</a>
+:marked
+  ### Suffixes
+  <a id="104"></a>
+  #### Style 104
+
+    - Append the component name with the suffix `Component`.
+
+      **Why?** The `Component` suffix is more commonly used and is more explicitly descriptive.
+
+      ```typescript
+      // recommended
+
+      // speaker-list.component.ts
+      export class SpeakerListComponent { }
+      ```
+
+<a href="#toc">Back to top</a>
+:marked
+  ### Service Names
+  <a id="110"></a>
+  #### Style 110
+
+    - Use consistent names for all services named after their feature. Use UpperCamelCase for services. Suffix services with `Service` when it is not clear what they are (e.g. when they are nouns).
+
+      **Why?** Provides a consistent way to quickly identify and reference services.
+
+      **Why?** Clear service names such as `logger` do not require a suffix.
+
+      **Why?** Service names such as `Credit` are nouns and require a suffix and should be named with a suffix when it is not obvious if it is a service or something else. 
+
+      ```typescript
+      SpeakerService // speaker.service.ts
+      CreditService  // credit.service.ts
+      Logger         // logger.service.ts
+      ```
+      
+<a href="#toc">Back to top</a>
+:marked
+  ## Routing
+  Client-side routing is important for creating a navigation flow between a component tree hierarchy, and composing components that are made of many other child components.
+
+<a href="#toc">Back to top</a>
+:marked
+  ### Component Router
+  <a id="130"></a>
+  #### Style 130
+
+    - Separate route configuration into a routing component file, also known as a component router.
+
+    - Use a `<router-outlet>` in the component router, where the routes will have their component targets display their templates.
+    
+    - Focus the logic in the component router to the routing aspects and its target components. Extract other logic to services and other components.
+    
+      **Why?** A component that handles routing is known as the component router, thus this follows the Angular 2 routing pattern.
+      
+      **Why?** A component that handles routing is known as the componenter router.
+
+      **Why?** The `<router-outlet>` indicates where the tempalte should be displayed for the target route.
+      
+      ```typescript
+      import { Component } from 'angular2/core';
+      import { RouteConfig, ROUTER_DIRECTIVES, ROUTER_PROVIDERS } from 'angular2/router';
+
+      import { SpeakersComponent, SpeakerService } from './+speakers';
+      import { DashboardComponent } from './+dashboard';
+      import { NavComponent } from './layout/nav.component';
+
+      @Component({
+        selector: 'my-app',
+        templateUrl: 'app/app.component.html',
+        styleUrls: ['app/app.component.css'],
+        directives: [ROUTER_DIRECTIVES, NavComponent],
+        providers: [
+          ROUTER_PROVIDERS,
```

Comment:
tldr; we try not to provide anything at boostrap. 

this makes it more difficult to test and often the root component is what really needs the providers, as it is the root of the component tree. this also makes the bootstrap super simple and focuses just on the platform.

i know @wardbell had some chats with Misko on this and he can chime in too.